src/components/cesium_map/api/spatial.js

```diff
@@ -212,24 +211,36 @@ export class ISamplesSpatial {
    * Create a new viewer
    * @param element Element or elementId
    */
-  constructor(element, initialLocation) {
+  constructor(element) {
     this.tracking_info = {
       color: Cesium.Color.BLUE,
       width: 10,
       tracking: false,
       polyline: null,
       positions: [],
     };
-    this.worldTerrain = Cesium.createWorldTerrain({});
-
     this.viewer = new Cesium.Viewer(element, {
       timeline: false,
       animation: false,
       sceneModePicker: false,
-      terrainProvider: this.worldTerrain,
       fullscreenElement: element
     });
+    this.osmBuildingsTileset = null;
+  }
 
+  async initialize() {
+    try {
+      this.worldTerrain = await Cesium.createWorldTerrainAsync();
+      this.osmBuildingsTileset = await Cesium.createOsmBuildingsAsync();
+      this.viewer.scene.primitives.add(this.osmBuildingsTileset);
+    } catch(error) {
```

Comment:
These two methods have changed to use the `async/await` pattern. As the constructor cannot be async, extract it outside to this `initialize()` function.